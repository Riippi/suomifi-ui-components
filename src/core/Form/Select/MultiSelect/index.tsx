export {
  MultiSelect,
  MultiSelectProps,
  MultiSelectData,
} from './MultiSelect/MultiSelect';
