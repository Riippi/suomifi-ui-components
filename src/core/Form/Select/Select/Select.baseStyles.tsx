import { css } from 'styled-components';
import { SuomifiTheme } from '../../../theme';
import { font, element } from '../../../theme/reset';

export const baseStyles = (theme: SuomifiTheme) => css`
  &.fi-select {
    ${font(theme)('bodyText')}
    width: 290px;

    & .fi-filter-input_input-element-container {
      position: relative;

      &:before {
        content: '';
        position: absolute;
        top: 50%;
        right: 16px;
        margin-top: -3px;
        border-style: solid;
        border-color: ${theme.colors.depthDark3} transparent transparent
          transparent;
        border-width: 6px 4px 0 4px;
        pointer-events: none;
      }
    }

    & .fi-filter-input_input {
      padding-right: 40px;
    }

    &--value-selected {
      & .fi-filter-input_input {
        padding-right: 64px;
      }
    }
  }

  & .fi-select_content_wrapper {
    display: inline-block;
    width: 100%;
    position: relative;
    & .fi-select_selected-value {
      position: absolute;
      top: 38px;
      ${element(theme)}
      ${font(theme)('actionElementInnerText')}
      max-width: 100%;
      padding: ${theme.spacing.insetM} 11px;
      line-height: 1;
    }
    & .fi-select_clear-button {
      position: absolute;
      top: 43px;
      right: 38px;
    }
  }

  &.fi-select--open {
    & .fi-filter-input_input-element-container {
      &:before {
        border-color: transparent transparent ${theme.colors.depthDark3}
          transparent;
        border-width: 0 4px 6px 4px;
      }
    }
    & .fi-filter-input_input {
      border-bottom: 0;
      border-bottom-left-radius: 0;
      border-bottom-right-radius: 0;
    }
  }
`;
