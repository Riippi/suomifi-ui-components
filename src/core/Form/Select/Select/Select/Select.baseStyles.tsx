import { css } from 'styled-components';
import { SuomifiTheme } from '../../../../theme';
import { font, element } from '../../../../theme/reset';

export const baseStyles = (theme: SuomifiTheme) => css`
  &.fi-select {
    ${font(theme)('bodyText')}
    width: 290px;

    & .fi-filter-input_input {
      padding-right: 38px;
    }

    &--value-selected {
      & .fi-filter-input_input {
        padding-right: 75px;
      }
    }
  }

  & .fi-select_content_wrapper {
    display: inline-block;
    width: 100%;
    position: relative;
    & .fi-select_selected-value {
      position: absolute;
      top: 38px;
      ${element(theme)}
      ${font(theme)('actionElementInnerText')}
      max-width: 100%;
      padding: ${theme.spacing.insetM} 11px;
      line-height: 1;
    }
    & .fi-select_clear-button_wrapper {
      border-right: 1px solid ${theme.colors.depthBase};
      width: 37px;
      height: 20px;
      top: 43px;
      right: 38px;
      position: absolute;
      & .fi-select_clear-button {
        position: absolute;
        top: 0;
        right: 8px;
      }
    }
    & .fi-select_toggle-button {
      position: absolute;
      top: 43px;
      right: 0;
    }
  }

  &.fi-select--open {
    & .fi-filter-input_input {
      border-bottom: 0;
      border-bottom-left-radius: 0;
      border-bottom-right-radius: 0;
    }
  }
`;
