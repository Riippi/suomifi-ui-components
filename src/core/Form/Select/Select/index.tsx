export { Select, SelectProps, SelectData } from './Select/Select';
