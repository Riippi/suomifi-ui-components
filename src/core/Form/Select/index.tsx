export { Select, SelectProps, SelectData } from './Select/Select';
export { MultiSelect, MultiSelectProps, MultiSelectData } from './MultiSelect';
