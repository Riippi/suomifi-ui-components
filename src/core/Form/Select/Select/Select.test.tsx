import React from 'react';
import { render, act } from '@testing-library/react';
import { axeTest } from '../../../../utils/test/axe';
import { Select } from './Select';

const tools = [
  {
    name: 'Jackhammer',
    price: 230,
    tax: false,
    labelText: 'Jackhammer',
    uniqueItemId: 'jh2435626',
  },
  {
    name: 'Hammer',
    price: 15,
    tax: true,
    labelText: 'Hammer',
    uniqueItemId: 'h9823523',
  },
  {
    name: 'Sledgehammer',
    price: 36,
    tax: false,
    labelText: 'Sledgehammer',
    uniqueItemId: 'sh908293482',
  },
  {
    name: 'Spade',
    price: 50,
    tax: true,
    labelText: 'Spade',
    uniqueItemId: 's82502335',
  },
  {
    name: 'Powersaw',
    price: 150,
    tax: false,
    labelText: 'Powersaw',
    disabled: true,
    uniqueItemId: 'ps9081231',
  },
  {
    name: 'Shovel',
    price: 115,
    tax: true,
    labelText: 'Shovel',
    uniqueItemId: 's05111511',
  },
  {
    name: 'Iron stick',
    price: 85,
    tax: false,
    labelText: 'Iron stick',
    uniqueItemId: 'is3451261',
  },
  {
    name: 'Rake',
    price: 50,
    tax: true,
    labelText: 'Rake',
    uniqueItemId: 'r09282626',
  },
  {
    name: 'Motorsaw',
    price: 450,
    tax: false,
    labelText: 'Motorsaw',
    disabled: true,
    uniqueItemId: 'ms6126266',
  },
];

const defaultSelectedTool = {
  name: 'Hammer',
  price: 15,
  tax: true,
  labelText: 'Hammer',
  uniqueItemId: 'h9823523',
};

const BasicCombobox = (
  <Select
    labelText="MultiSelect"
    items={tools}
    visualPlaceholder="Choose your tool(s)"
    noItemsText="No items"
    defaultSelectedItem={defaultSelectedTool}
  />
);

it('should not have basic accessibility issues', async () => {
  await act(async () => {
    axeTest(BasicCombobox);
  });
});

it('has matching snapshot', () => {
  const { container } = render(BasicCombobox);
  expect(container.firstChild).toMatchSnapshot();
});

// describe('Controlled', () => {
//   it('has the controlled items as selected + disabled', async () => {
//     const controlledItems = [
//       {
//         name: 'Shovel',
//         price: 115,
//         tax: true,
//         labelText: 'Shovel',
//         uniqueItemId: 's05111511',
//         disabled: true,
//       },
//       {
//         name: 'Sledgehammer',
//         price: 36,
//         tax: false,
//         labelText: 'Sledgehammer',
//         uniqueItemId: 'sh908293482',
//       },
//     ];
//     const multiselect = (
//       <Select
//         selectedItems={controlledItems}
//         labelText="MultiSelect"
//         items={tools}
//         chipListVisible={true}
//         ariaChipActionLabel="Remove"
//         removeAllButtonLabel="Remove all selections"
//         visualPlaceholder="Choose your tool(s)"
//         noItemsText="No items"
//         defaultSelectedItems={defaultSelectedTools}
//         ariaSelectedAmountText="tools selected"
//       />
//     );

//     await act(async () => {
//       const { container } = render(multiselect);
//       expect(container.querySelectorAll('.fi-chip').length).toEqual(2);

//       const chips = container.querySelectorAll('.fi-chip');

//       const disabledChip = chips[0];
//       expect(disabledChip).toHaveTextContent('Shovel');
//       expect(disabledChip).toHaveClass('fi-chip--disabled');
//       expect(disabledChip).toHaveAttribute('aria-disabled');

//       const otherChip = chips[1];
//       expect(otherChip).toHaveTextContent('Sledgehammer');
//     });
//   });

//   it('does not allow removing of items by clicking', async () => {
//     const animals = [
//       {
//         age: 2,
//         labelText: 'Rabbit',
//         uniqueItemId: 'rabbit-123',
//       },
//       {
//         age: 1,
//         labelText: 'Snail',
//         uniqueItemId: 'snail-321',
//       },
//       {
//         price: 5,
//         labelText: 'Turtle',
//         uniqueItemId: 'turtle-987',
//       },
//     ];

//     const multiselect = (
//       <Select
//         items={animals}
//         selectedItems={[
//           { price: 5, labelText: 'Turtle', uniqueItemId: 'turtle-987' },
//         ]}
//         labelText="Animals"
//         noItemsText="No animals"
//         chipListVisible={true}
//         visualPlaceholder="Try to choose animal(s)"
//         ariaChipActionLabel="Remove"
//         ariaSelectedAmountText="tools selected"
//       />
//     );

//     const { getByText, getAllByText } = render(multiselect);
//     const turtleChip = getByText('Turtle');
//     await act(async () => {
//       fireEvent.click(turtleChip, {});
//     });
//     // Popover is open, so therefore two
//     expect(getAllByText('Turtle').length).toBe(2);
//   });
// });

// it('should have correct baseClassName', async () => {
//   await act(async () => {
//     const { container } = render(BasicCombobox);
//     expect(container.firstChild).toHaveClass('fi-multiselect');
//   });
// });

// test('className: has given custom classname', async () => {
//   await act(async () => {
//     const { container } = render(
//       <Select
//         labelText="MultiSelect"
//         items={[]}
//         noItemsText="No items"
//         className="custom-class"
//         ariaSelectedAmountText=""
//       />,
//     );
//     expect(container.firstChild).toHaveClass('custom-class');
//   });
// });

// test('labelText: has the given text as label', async () => {
//   await act(async () => {
//     const { queryByText } = render(
//       <Select
//         labelText="MultiSelect"
//         items={[]}
//         noItemsText="No items"
//         ariaSelectedAmountText=""
//       />,
//     );
//     expect(queryByText('MultiSelect')).not.toBeNull();
//   });
// });

// test('visualPlaceholder: has the given text as placeholder attribute', () => {
//   const { getByRole } = render(
//     <Select
//       labelText="MultiSelect"
//       items={[]}
//       noItemsText="No items"
//       visualPlaceholder="Select item(s)"
//       ariaSelectedAmountText=""
//     />,
//   );
//   const inputfield = getByRole('textbox') as HTMLInputElement;
//   expect(inputfield).toHaveAttribute('placeholder', 'Select item(s)');
// });

// test('id: has the given id', () => {
//   const { getByRole } = render(
//     <Select
//       id="cb-123"
//       labelText="MultiSelect"
//       items={[]}
//       noItemsText="No items"
//       ariaSelectedAmountText=""
//     />,
//   );
//   expect(getByRole('textbox')).toHaveAttribute('id', 'cb-123');
// });

// describe('statusText', () => {
//   it('should have element and correct classname for it', () => {
//     const { getByText } = render(
//       <Select
//         id="123"
//         labelText="MultiSelect"
//         items={[]}
//         noItemsText="No items"
//         visualPlaceholder="Select item(s)"
//         statusText="EROR EROR"
//         ariaSelectedAmountText=""
//       />,
//     );
//     const statusText = getByText('EROR EROR');
//     expect(statusText).toHaveClass('fi-status-text');
//   });

//   it('will be added to input aria-describedby', () => {
//     const { getByRole } = render(
//       <Select
//         id="123"
//         labelText="MultiSelect"
//         items={[]}
//         noItemsText="No items"
//         visualPlaceholder="Select item(s)"
//         statusText="EROR EROR"
//         ariaSelectedAmountText=""
//       />,
//     );
//     expect(getByRole('textbox')).toHaveAttribute(
//       'aria-describedby',
//       '123-statusText 123-selectedItems-length',
//     );
//   });
// });

// describe('status', () => {
//   it('should have error classname', () => {
//     const { container } = render(
//       <Select
//         id="123"
//         labelText="MultiSelect"
//         items={[]}
//         noItemsText="No items"
//         visualPlaceholder="Select item(s)"
//         status="error"
//         ariaSelectedAmountText=""
//       />,
//     );
//     expect(container.firstChild).toHaveClass('fi-multiselect--error');
//   });
// });
