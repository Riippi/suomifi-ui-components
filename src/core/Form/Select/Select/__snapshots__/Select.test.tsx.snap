// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has matching snapshot 1`] = `
.c7 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  overflow: visible;
  text-transform: none;
  -webkit-appearance: button;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline-block;
  max-width: 100%;
  cursor: pointer;
}

.c7:-moz-focusring {
  outline: 1px dotted ButtonText;
}

.c7::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.c7::-webkit-file-upload-button {
  -webkit-appearance: button;
  font: inherit;
}

.c7::-webkit-inner-spin-button {
  height: auto;
}

.c7::-webkit-outer-spin-button {
  height: auto;
}

.c7::before,
.c7::after {
  box-sizing: border-box;
}

.c0 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c0::before,
.c0::after {
  box-sizing: border-box;
}

.c5 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  overflow: visible;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline-block;
  max-width: 100%;
}

.c5::-webkit-input-placeholder {
  color: inherit;
  opacity: 0.54;
}

.c5::before,
.c5::after {
  box-sizing: border-box;
}

.c4 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c4::before,
.c4::after {
  box-sizing: border-box;
}

.c9 {
  position: absolute;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  width: 1px;
  margin: -1px;
  padding: 0;
  border: 0;
  overflow: hidden;
}

.c3.fi-label-text .fi-label-text_label-span {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 600;
  display: block;
  margin-bottom: 10px;
  color: hsl(0,0%,16%);
}

.c3.fi-label-text .fi-label-text_label-span .fi-label-text_optionalText {
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
}

.c6.fi-status-text {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 600;
  color: hsl(0,0%,16%);
  font-size: 14px;
  line-height: 20px;
}

.c6.fi-status-text.fi-status-text--error {
  color: hsl(3,59%,48%);
}

.c6.fi-status-text.fi-status-text--hasContent {
  margin-top: 5px;
}

.c2 .fi-filter-input {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
}

.c2 .fi-filter-input_wrapper {
  display: inline-block;
  width: 100%;
}

.c2 .fi-filter-input_functionalityContainer {
  line-height: 1.5;
}

.c2 .fi-filter-input_input-element-container > input:focus {
  outline-color: hsl(196,77%,44%);
  outline-width: 2px;
  outline-offset: 2px;
  outline-style: solid;
}

.c2 .fi-filter-input_input-element-container:focus-within > input:focus {
  outline: none;
}

.c2 .fi-filter-input_input-element-container:focus-within {
  position: relative;
}

.c2 .fi-filter-input_input-element-container:focus-within::after {
  content: '';
  position: absolute;
  pointer-events: none;
  top: -2px;
  right: -2px;
  bottom: -2px;
  left: -2px;
  border-radius: 2px;
  background-color: transparent;
  border: 0px solid hsl(0,0%,100%);
  box-sizing: border-box;
  box-shadow: 0 0 0 2px hsl(196,77%,44%);
  z-index: 9999;
}

.c2 .fi-filter-input_input {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  min-width: 245px;
  max-width: 100%;
  padding: 8px 16px;
  border: 1px solid hsl(202,7%,80%);
  border-radius: 2px;
  line-height: 1;
  background-color: hsl(0,0%,100%);
  min-width: 40px;
  width: 100%;
  min-height: 40px;
  padding-left: 10px;
  border-color: hsl(201,7%,58%);
}

.c2 .fi-filter-input_input::-webkit-input-placeholder {
  font-style: italic;
  color: hsl(201,7%,46%);
  opacity: 1;
}

.c2 .fi-filter-input_input::-moz-placeholder {
  font-style: italic;
  color: hsl(201,7%,46%);
  opacity: 1;
}

.c2 .fi-filter-input_input:-ms-input-placeholder {
  font-style: italic;
  color: hsl(201,7%,46%);
  opacity: 1;
}

.c2 .fi-filter-input_input::placeholder {
  font-style: italic;
  color: hsl(201,7%,46%);
  opacity: 1;
}

.c2.fi-filter-input--error .fi-filter-input_input {
  border-color: hsl(3,59%,48%);
  border-width: 2px;
}

.c2.fi-filter-input--disabled .fi-filter-input_input {
  color: hsl(202,7%,67%);
  background-color: hsl(202,7%,97%);
  border-color: hsl(202,7%,80%);
}

.c2.fi-filter-input--label-align-left .fi-filter-input_wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2.fi-filter-input--label-align-left .fi-filter-input_wrapper .fi-label-text {
  padding-right: 10px;
  padding-top: 8px;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
}

.c2.fi-filter-input--label-align-left .fi-filter-input_wrapper .fi-label-text .fi-label-text_label-span {
  margin-bottom: 0;
}

.c10 {
  display: inline-block;
  vertical-align: baseline;
}

.c10.fi-icon--cursor-pointer {
  cursor: pointer;
}

.c10.fi-icon--cursor-pointer * {
  cursor: inherit;
}

.c8.fi-clear-button {
  cursor: pointer;
  pointer-events: all;
  height: 20px;
  width: 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8.fi-clear-button:focus {
  outline: none;
}

.c8.fi-clear-button:focus:after {
  content: '';
  position: absolute;
  pointer-events: none;
  top: -2px;
  right: -2px;
  bottom: -2px;
  left: -2px;
  border-radius: 2px;
  background-color: transparent;
  border: 0px solid hsl(0,0%,100%);
  box-sizing: border-box;
  box-shadow: 0 0 0 2px hsl(196,77%,44%);
  z-index: 9999;
}

.c8.fi-clear-button .fi-clear-button_icon {
  width: 16px;
  height: 16px;
  fill: hsl(212,63%,37%);
}

.c1.fi-select {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
  width: 290px;
}

.c1.fi-select .fi-filter-input_input-element-container {
  position: relative;
}

.c1.fi-select .fi-filter-input_input-element-container:before {
  content: '';
  position: absolute;
  top: 50%;
  right: 16px;
  margin-top: -3px;
  border-style: solid;
  border-color: hsl(201,7%,58%) transparent transparent transparent;
  border-width: 6px 4px 0 4px;
  pointer-events: none;
}

.c1.fi-select .fi-filter-input_input {
  padding-right: 40px;
}

.c1.fi-select--value-selected .fi-filter-input_input {
  padding-right: 64px;
}

.c1 .fi-select_content_wrapper {
  display: inline-block;
  width: 100%;
  position: relative;
}

.c1 .fi-select_content_wrapper .fi-select_selected-value {
  position: absolute;
  top: 38px;
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
  max-width: 100%;
  padding: 8px 11px;
  line-height: 1;
}

.c1 .fi-select_content_wrapper .fi-select_clear-button {
  position: absolute;
  top: 43px;
  right: 38px;
}

.c1.fi-select--open .fi-filter-input_input-element-container:before {
  border-color: transparent transparent hsl(201,7%,58%) transparent;
  border-width: 0 4px 6px 4px;
}

.c1.fi-select--open .fi-filter-input_input {
  border-bottom: 0;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
}

<div
  aria-expanded="false"
  aria-haspopup="listbox"
  aria-owns="2-popover"
  class="c0 fi-select c1 fi-select--value-selected"
  role="combobox"
>
  <div
    class="c0 fi-select_content_wrapper"
  >
    <div
      class="c0 fi-filter-input c2"
    >
      <div
        class="c0 fi-filter-input_wrapper"
      >
        <label
          class="c0 c3 fi-label-text"
          for="1"
        >
          <span
            class="c4 fi-label-text_label-span"
          >
            MultiSelect
          </span>
        </label>
        <div
          class="c0 fi-filter-input_functionalityContainer"
        >
          <div
            class="c0 fi-filter-input_input-element-container"
          >
            <input
              aria-activedescendant=""
              aria-autocomplete="list"
              aria-controls="2-popover"
              aria-describedby="2-aria-status"
              aria-multiline="false"
              autocapitalize="none"
              autocomplete="off"
              class="c5 fi-filter-input_input"
              id="1"
              placeholder=""
              readonly=""
              spellcheck="false"
              type="text"
              value="Hammer"
            />
          </div>
          <span
            aria-atomic="true"
            aria-live="assertive"
            class="c4 c6 fi-status-text"
          />
        </div>
      </div>
    </div>
    <button
      class="c7 fi-clear-button c8 fi-select_clear-button"
      type="button"
    >
      <span
        class="c4 c9 fi-visually-hidden"
      >
        Clear selection
      </span>
      <svg
        aria-hidden="true"
        class="fi-icon c10 fi-clear-button_icon"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 13.45L4.75 20.7a1.026 1.026 0 01-1.45-1.45L10.55 12 3.3 4.75A1.026 1.026 0 014.75 3.3L12 10.55l7.25-7.25a1.026 1.026 0 011.45 1.45L13.45 12l7.25 7.25a1.026 1.026 0 01-1.45 1.45L12 13.45z"
        />
      </svg>
    </button>
  </div>
  <span
    class="c4 c9 fi-visually-hidden"
  >
    <span
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="all"
      class="c4"
      id="2-aria-status"
    />
  </span>
</div>
`;
